spec/scheduling-tasks.md

```diff
@@ -12,15 +12,17 @@ This spec formalizes three priorities to support scheduling tasks:
   };
 </pre>
 
-<dfn enum-value for=TaskPriority>user-blocking</dfn> is the highest priority, and is meant to be
-used for tasks that are blocking the user's ability to interact with the page, such as rendering the
-core experience or responding to user input.
+<dfn enum-value for=TaskPriority>user-blocking</dfn> is the highest priority, and it is meant for
+tasks that should run as soon as possible, such that running them at a lower priority would degrade
+user experience. This could be (chunked) work that is directly in response to user input, or
```

Comment:
"work that is directly in response to user input" I would naively interpret as "needed before next paint", which this priority would not guarantee.

I know you are trying to imply this is important follow-up after initial feedback to user input ((via the "chunked" reference), but I would really consider being even more explicit about it, as only influencing "relatively likelihood", but that no task scheduling priority has any guarantees, or something.

It seems likely to me that folks would actually default assume "user-blocking" means render-blocking (next paint), especially in contrast to the naming of "user-visible" (if both are for visible input, what possible difference could there be?).

I also assume most folks (maybe not via reading spec, but via articles copying this list) won't read the algorithms just the descriptions.

Reply:
Thanks, good point -- and I forgot that I had added something like that to the meta-explainer based on other feedback. I added a paragraph to basically say user-blocking != render-blocking.